### PR TITLE
Add per method retries

### PR DESCRIPTION
**Pull request by nspragg**
* Adds support for configuring retries via method options

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -147,9 +147,9 @@ Client.prototype._requestWithRetries = function (method, url, opts, cb) {
   var client = this;
 
   var operation = retry.operation({
-    retries: this.retries,
-    minTimeout: this.retryTimeout,
-    maxTimeout: this.retryTimeout
+    retries: opts.retries || this.retries,
```
Comment (robinjmurphy) [inline lib/client.js]: This won't work if retries is set to `0`.

Comment (robinjmurphy) [inline lib/client.js]: Might need to use `_.isUndefined`

File: test/client.js [hunk 2]
```diff
@@ -483,12 +496,26 @@ describe('Rest Client', function () {
         retries: 2,
         retryTimeout: 0
       });
-      api.put(path, requestBody).reply(500, responseBody);
-      api.put(path, requestBody).reply(500, responseBody);
+      nockRetries(2, {
+        httpMethod: 'put'
+      });
+
       api.put(path, requestBody).reply(201, responseBody);
```
Comment (robinjmurphy) [inline test/client.js]: Don't think the final response code is necessary. You can just make use of your new `successCode: 201` property.

File: test/client.js [hunk 3]
```diff
@@ -591,8 +632,8 @@ describe('Rest Client', function () {
       var statusCode = 418;
       api.post(path, requestBody).reply(statusCode);
       client.post(url, requestBody, function (err, body, resp) {
-          assert.equal(resp.statusCode, statusCode);
-          done();
+        assert.equal(resp.statusCode, statusCode);
```
Comment (robinjmurphy) [inline test/client.js]: Bit of boy scouting... can we add an `assert.ifError` here?

Comment (nspragg) [inline test/client.js]: Can add a assert(err). We don't want an assert.ifError() as it's supposed to be an error in this case (418)

Comment (robinjmurphy) [inline test/client.js]: Ahh my bad. Don't worry.

File: lib/client.js (hunk 1 above)
Comment (nspragg) [inline lib/client.js]: Good spot - will fix

File: lib/client.js [hunk 4]
```diff
@@ -146,10 +146,11 @@ Client.prototype.getWithResponse = function (url, opts, cb) {
 Client.prototype._requestWithRetries = function (method, url, opts, cb) {
   var client = this;
 
+  var timeout = _.isUndefined(opts.retryTimeout) ? this.retryTimeout : opts.retryTimeout;
   var operation = retry.operation({
-    retries: this.retries,
-    minTimeout: this.retryTimeout,
-    maxTimeout: this.retryTimeout
+    retries: _.isUndefined(opts.retries) ? this.retries : opts.retries,
```
Comment (robinjmurphy) [inline lib/client.js]: Being super picky now, but would be good to put this into a variable too `retries`.

**Comment by robinjmurphy** [comment]
:+1:

**Pull request merged**